cli/cmd/install/install.go

```diff
@@ -0,0 +1,500 @@
+	flagNameFileValues      = "set-file"
+
+	flagNameDryRun = "dry-run"
+	defaultDryRun  = false
+
+	flagNameAutoApprove = "auto-approve"
+	defaultAutoApprove  = false
+
+	flagNameNamespace = "namespace"
+	defaultNamespace  = "consul"
+
+	helmRepository = "https://helm.releases.hashicorp.com"
+)
+
+type Command struct {
+	*common.BaseCommand
+
+	kubernetes kubernetes.Interface
+
+	set *flag.Sets
+
+	flagPreset          string
+	flagNamespace       string
+	flagDryRun          bool
+	flagAutoApprove     bool
+	flagValueFiles      []string
+	flagSetStringValues []string
+	flagSetValues       []string
+	flagFileValues      []string
+
+	flagKubeConfig  string
+	flagKubeContext string
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	// Store all the possible preset values in 'presetList'. Printed in the help message.
+	var presetList []string
+	for name := range presets {
+		presetList = append(presetList, name)
+	}
+
+	c.set = flag.NewSets()
+	{
+		f := c.set.NewSet("Command Options")
+		f.BoolVar(&flag.BoolVar{
+			Name:    flagNameAutoApprove,
+			Target:  &c.flagAutoApprove,
+			Default: defaultAutoApprove,
+			Usage:   "Skip confirmation prompt.",
+		})
+		f.BoolVar(&flag.BoolVar{
+			Name:    flagNameDryRun,
+			Target:  &c.flagDryRun,
+			Default: defaultDryRun,
+			Usage:   "Validate installation and return summary of installation.",
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:    flagNameConfigFile,
+			Aliases: []string{"f"},
+			Target:  &c.flagValueFiles,
+			Usage:   "Path to a file to customize the installation, such as Consul Helm chart values file. Can be specified multiple times.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    flagNameNamespace,
+			Target:  &c.flagNamespace,
+			Default: defaultNamespace,
+			Usage:   fmt.Sprintf("Namespace for the Consul installation. Defaults to \"%q\".", defaultNamespace),
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    flagNamePreset,
+			Target:  &c.flagPreset,
+			Default: defaultPreset,
+			Usage:   fmt.Sprintf("Use an installation preset, one of %s. Defaults to \"%q\"", strings.Join(presetList, ", "), defaultPreset),
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   flagNameSetValues,
+			Target: &c.flagSetValues,
+			Usage:  "Set a value to customize. Can be specified multiple times. Supports Consul Helm chart values.",
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   flagNameFileValues,
+			Target: &c.flagFileValues,
+			Usage: "Set a value to customize via a file. The contents of the file will be set as the value. Can be " +
+				"specified multiple times. Supports Consul Helm chart values.",
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   flagNameSetStringValues,
+			Target: &c.flagSetStringValues,
+			Usage:  "Set a string value to customize. Can be specified multiple times. Supports Consul Helm chart values.",
+		})
+
+		f = c.set.NewSet("Global Options")
+		f.StringVar(&flag.StringVar{
```

Comment:
Given that by default people keep the kubeconfig file at `~/.kube/config`, I think it would make sense to use that as the default.

Reply:
In the default case, for any command that uses helm, we'll want to respect Helm's order of evaluation to get the kube context. I think if we were to explicitly set this to ~/.kube/config, then we wouldn't be able to respect the Helm library's "backups" for finding a kubeconfig (i.e an env var, etc.).

Reply:
Cool, then this is just me not knowing yet :) Thanks!